test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -353,6 +352,41 @@ public void UseEnvironmentIsNotOverriden()
         }
 
         [Fact]
+        public void EnvironmentConfigurationFallbackHierarchyHonored()
+        {
+            var testServer = new TestServer();
+
+            // Default to Production
+            var host = new WebHostBuilder()
+                .UseServer(testServer)
+                .UseStartup("Microsoft.AspNetCore.Hosting.Tests")
+                .Build();
+
+            Assert.Equal(EnvironmentName.Production, host.Services.GetService<IHostingEnvironment>().EnvironmentName);
+
+            // Use environment variables if available
+            Environment.SetEnvironmentVariable(WebHostDefaults.EnvironmentVariablesPrefix + WebHostDefaults.EnvironmentKey, "EnvironmentA");
```

Comment:
They do, though they will only fail if the deployed test tests something that's environment specific and the failure can be hard to trace back to this issue? Also, I don't think anything tests the hierarchy of the fallbacks, but I assume that it's not deemed important.